packages/babel-plugin-proposal-decorators/CONTRIB.md

```diff
@@ -0,0 +1,374 @@
+this.
+
+Overall, this decoration is pretty straightforward other than the fact that we
+have to reference `initA` externally.
+
+```js
+  static #b(...args) {
+    callB(this, args);
+  }
+  static x() {
+    console.log('foo');
+  }
+```
+
+Next up, we have a private static class method `#b`. This one is a bit more
+complex, as our definition has been broken out into 2 parts:
+
+1. `static #b`: This is the method itself, which being a private method we
+  cannot overwrite with `defineProperty`. We also can't convert it into a
+  private field because that would change its semantics (would make it
+  writable). So, we instead have it proxy to the locally scoped `callB`
+  variable, which will be populated with the fully decorated method.
+2. `static x`: This contains the _code_ of the original method. Once again, this
+  code cannot be removed from the class body because it may reference private
+  identifiers. However, we have moved the code to a _public_ method, which means
+  we can now read its value using `Object.getOwnPropertyDescriptor`. Decorators
+  use this to get the initial implementation of the method, which can then be
+  wrapped with decorator code/logic. They then `delete` this temporary property,
+  which is necessary because no additional elements should be added to a class
+  definition.
+
+  The name for this method is unimportant, but because public method names
+  cannot be minified and we also need to pass the name into `applyDecs`, we
+  generate as small of a unique identifier as possible here, starting with 1
+  char names which are not taken and growing until we find one that is free.
+
+```js
+  [computedKey]() {}
+```
+
+Next is the undecorated method with a computed key. This uses the previously
+calculated and stored computed key.
+
+```js
+  #y = initC(this, 123);
+  get y() {
+    return this.#y;
+  }
+  set y(v) {
+    this.#y = v;
+  }
+  get #c() {
+    return getC(this);
+  }
+  set #c(v) {
+    setC(this, v);
+  }
+```
+
+Next up, we have the output for `accessor #c`. This is the most complicated
+case, since we have to transpile the decorators, the `accessor` keyword, and
+target a private field. Breaking it down piece by piece:
+
+```js
+  #y = initC(this, 123);
+```
+
+`accessor #c` desugars to a getter and setter which are backed by a new private
+field, `#y`. Like before, the name of this field doesn't really matter, we'll
+just generate a short, unique name. We call the decorated initializer for `#c`
+and return that value to assign to the field.
+
+```js
+  get y() {
+    return this.#y;
+  }
+  set y(v) {
+    this.#y = v;
+  }
+```
+
+Next we have a getter and setter named `y` which provide access to the backing
+storage for the accessor. These are the base getter/setter for the accessor,
+which the decorator will get via `Object.getOwnPropertyDescriptor`. They will be
+deleted from the class fully, so again the name is not important here, just
```

Comment:
Also here, if possible I'd prefer to pass them in a new last param of `applyDecs` rather than deleting.